foundation/src/main/java/org/researchstack/foundation/core/models/task/Task.java

```diff
@@ -13,19 +15,24 @@
  * <p>
  * To present the ResearchStack framework UI in your app, instantiate an object that extends the
  * Task class (such as {@link OrderedTask}) and provide it to a {@link
- * org.researchstack.backbone.ui.ViewTaskActivity}.
+ * org.researchstack.foundation.ui.ViewTaskActivity}.
  * <p>
  * Implement this protocol to enable dynamic selection of the steps for a given task. By default,
  * OrderedTask implements this protocol for simple sequential tasks.
  * <p>
  * Each {@link Step} in a task roughly corresponds to one screen, and represents the primary unit of
- * work in any task. For example, a {@link org.researchstack.backbone.step.QuestionStep} object
+ * work in any task. For example, a {@link org.researchstack.foundation.step.QuestionStep} object
  * corresponds to a single question presented on screen, together with controls the participant uses
- * to answer the question. Another example is {@link org.researchstack.backbone.step.FormStep},
+ * to answer the question. Another example is {@link org.researchstack.foundation.step.FormStep},
  * which corresponds to a single screen that displays multiple questions or items for which
  * participants provide information, such as first name, last name, and birth date.
  */
-public abstract class Task implements Serializable {
+public abstract class Task implements Serializable, ITask, ITaskNavigator<Step, TaskResult> {
+
+    public interface TaskLifecycleObserver {
```

Comment:
Why would you have an empty interface?

Reply:
I was running into an issue where there were circular dependencies between Task and ViewTaskActivity. In order to remove the dependency that Task had on ViewTaskActivity, I implemented this interface.